http-body-util/src/combinators/collect.rs

```diff
@@ -37,7 +38,10 @@ impl<T: Body + ?Sized> Future for Collect<T> {
                 return Poll::Ready(Ok(me.collected.take().expect("polled after complete")));
             };
 
-            me.collected.as_mut().unwrap().push_frame(frame);
+            // Only push this frame if it's a trailer frame, or if it has some data in it.
```

Comment:
Should probably only collect DATA frames, not trailers.

Reply:
Yeah, initially I added the `if` check just for `frame.data_ref`, but then the [`trailers` test](https://github.com/hyperium/http-body/blob/master/http-body-util/src/collected.rs#LL158C17-L158C17) fails with:


```sh
---- collected::tests::trailers stdout ----
thread 'collected::tests::trailers' panicked at 'called `Option::unwrap()` on a `None` value', http-body-util/src/collected.rs:157:51
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Any ideas why?

Reply:
Not sure why its failing it looks correct to me....are you still running into this issue?

Reply:
Oh I see, that's because `push_frame` determines how to store a given frame type. In that case, I'd move the check into that function, and in the branch that notices a data frame, add a check for the data being empty.

Reply:
Thanks for the explanation.

That's an even better solution.